### PR TITLE
Use monorepo plugin server

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

This opens the floodgates and starts using the local plugin-server dir instead of pulling the `posthog-plugin-server` package. We should only flip the switch when we're for sure ready. 

Let's probably merge #6737 first too

## How did you test this code?

*Please describe.*

### CODE REVIEW

**Comment by fuziontech** [comment]
wooops! wrong PR

(github UI has been very laggy for me recently)

**Review by fuziontech — APPROVED** [review]

**Comment by yakkomajuri** [comment]
Each time a new Cypress test fails :D

**Comment by yakkomajuri** [comment]
I guess updating some docker files would be needed here too 🤔

File: dev.Dockerfile [hunk 1]
```diff
@@ -62,15 +62,14 @@ RUN apk --update --no-cache --virtual .build-deps add \
...
-COPY plugins/package.json plugins/yarn.lock ./plugins/
+COPY ./plugin-server/ ./plugin-server/
 RUN apk --update --no-cache --virtual .build-deps add \
     "gcc~=10.3" \
     && \
     yarn config set network-timeout 300000 && \
     yarn install --frozen-lockfile && \
-    yarn install --frozen-lockfile --cwd plugins && \
-    yarn cache clean \
-    && \
```
Comment (guidoiaquinti) [inline dev.Dockerfile:73]: Nit: can we keep this `&&` visual separation between commands and dependencies ?

File: production.Dockerfile [hunk 2]
```diff
@@ -51,7 +51,8 @@ RUN yarn --frozen-lockfile
 # load entire codebase & build frontend
 COPY . /code/
 RUN yarn build \
-    && yarn --cwd plugins --frozen-lockfile --ignore-optional \
+    && yarn --cwd plugin-server --frozen-lockfile --ignore-optional \
+    && yarn build --cwd plugin-server \
```
Comment (guidoiaquinti) [inline production.Dockerfile:55]: Why is this additional build necessary?

Comment (yakkomajuri) [reply production.Dockerfile:55]: So before we would pull the plugin server npm package, which comes readily built. Now, we have the plugin server code in here so need to build it ourselves.

**Review by Twixes — COMMENTED** [review]
This _looks_ alright and works in dev, but not sure about production. Did you try spinning up a prod-like PostHog instance from this branch?

**Review by macobo — APPROVED** [review]

**Comment by yakkomajuri** [comment]
Turns out, despite the approvals, this was 💩 

I was doing some thorough checking before shipping and have made some updates now.

I also decided to split out a prod script to avoid having handling for prod vs dev in the same script that runs cloud.

Updated the "old" script for backwards compatibility with everything else that uses it. 

So, another look? @macobo @Twixes @fuziontech

File: task-definition.plugins.json [hunk 3]
```diff
@@ -135,7 +135,7 @@
                     "valueFrom": "arn:aws:secretsmanager:us-east-1:795637471508:secret:Posthog_Production_Heroku-FQ2itU:STATSD_HOST::"
                 }
             ],
-            "entryPoint": ["./bin/plugin-server", "--no-restart-loop"],
+            "entryPoint": ["./bin/plugin-server-prod"],
```
Comment (yakkomajuri) [inline task-definition.plugins.json:138]: important change

Comment (macobo) [reply task-definition.plugins.json:138]: Why change this? We'd end up needing to change _all_ of our deployments as well.

Comment (yakkomajuri) [reply task-definition.plugins.json:138]: Check how I made the original file backwards compatible too. So no need to change all deployments.

However, I'd like to start introducing this separation. Handling dev and prod on the same script is a bit iffy and I thought this was more clear

Comment (macobo) [reply task-definition.plugins.json:138]: Maybe let's introduce the new script into our prod + other deploys separately then? There's IMO risk involved with running plugin server differently in cloud and say helm - good way to introduce bugs, etc.

Comment (yakkomajuri) [reply task-definition.plugins.json:138]: sure, will revert this

Comment (yakkomajuri) [reply task-definition.plugins.json]: done, see if the new `bin/plugin-server` makes sense then

File: .dockerignore [hunk 4]
```diff
@@ -3,7 +3,7 @@
 !requirements.txt
 !requirements-dev.txt
 !posthog
-!plugins
+!plugin-server
```
Comment (macobo) [inline .dockerignore:6]: Not sure of the implications of this - e.g. when developing plugins w/ docker, will this cause issues?

**Review by macobo — APPROVED** [review]

File: .dockerignore (hunk 4 above)
Comment (yakkomajuri) [reply .dockerignore:6]: This was added to make sure we _can_ develop with Docker actually.

File: production.Dockerfile [hunk 5]
```diff
@@ -79,27 +79,35 @@ RUN apk --update --no-cache --virtual .build-deps add \
...
-    yarn install --frozen-lockfile --cwd plugins && \
+    yarn install --frozen-lockfile --cwd plugin-server && \
     yarn cache clean \
     && \
     apk del .build-deps
 
 # Copy everything else
 COPY . .
 
+
```
Comment (yakkomajuri) [inline production.Dockerfile:96]:
```suggestion
```

File: production.Dockerfile [hunk 6]
```diff
@@ -79,27 +79,35 @@ RUN apk --update --no-cache --virtual .build-deps add \
 # - we need few additional OS packages for this. Let's install
 #   and then uninstall them when the compilation is completed.
 COPY package.json yarn.lock ./
-COPY plugins/package.json plugins/yarn.lock ./plugins/
+COPY ./plugin-server/ ./plugin-server/
```
Comment (yakkomajuri) [inline production.Dockerfile:82]: I had some conflicts with you @guidoiaquinti so you might want to quickly skim this over

Comment (yakkomajuri) [reply production.Dockerfile:82]: the file I mean

**Comment by yakkomajuri** [comment]
Cypress failures actually look legit 🤔 Could indicate plugin server isn't running on those

**Comment by yakkomajuri** [comment]
We shouldn't delete `plugin-server/yarn.lock` @fuziontech

**Comment by yakkomajuri** [comment]
Also @fuziontech I'm reluctant to get your changes in. I also believe building should be done in Docker, but that needs us to do #7094 first, which I'd do in another PR. Not every script that runs /bin/plugin-server builds first, including our task definition for ECS, which relies on that script building it.

**Comment by fuziontech** [comment]
Seems reasonable. I'd like to get this in today to wrap this up. We can always refactor entry points and bin files later.

Environments I've tested this PR in:
- [x] Cloud - ECS (fixed with this PR https://github.com/PostHog/posthog-cloud/pull/149)
- [x] Docker (prod and dev)
- [x] Cypress
- [x] docker-compose
- [x] Helm
- [ ] dev - I've tested this in my environment but I don't know everyones.

**Pull request merged**